ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -407,21 +419,34 @@ function renderPodGroup(props: ApplicationResourceTreeProps, id: string, node: R
     const margin = 8;
     let topExtra = 0;
     const podGroup = node.podGroup;
+    const podGroupHealthy = podGroup.pods.filter(pod => pod.health === 'Healthy');
```

Comment:
modified the code to  include the check

```
   const podGroupHealthy = (podGroup.pods || []).filter(pod => pod.health === 'Healthy');
    const podGroupDegraded = (podGroup.pods || []).filter(pod => pod.health === 'Degraded');
    const podGroupInProgress = (podGroup.pods || []).filter(pod => pod.health === 'Progressing');
```

Reply:
Since podGroup is optional, I think `podGroup.pods` will still fail catastrophically.   Eg. Cannot read properties of null

 I think you still have to check if podGroup is not null.

Reply:
Added the check , but `renderPodGroup` will be called  when PodGroup flow is called